src/audio/mixer.c

```diff
@@ -737,6 +738,11 @@ static int mixin_bind(struct comp_dev *dev, void *data)
 			}
 		}
 
+		if (!sink_buf) {
+			comp_err(dev, "mixin_bind: no sink buffer found");
+			return -EINVAL;
+		}
```

Comment:
this makes it clear, that the case `dev->ipc_config.id != src_id` is actually invalid, so we could make that a direct test in the beginning of the function to make that even a bit more obvious